core/src/main/java/tc/oc/pgm/loot/FillerDefinition.java

```diff
@@ -0,0 +1,116 @@
+package tc.oc.pgm.loot;
+
+import java.time.Duration;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.bukkit.inventory.ItemStack;
+import tc.oc.pgm.api.feature.FeatureDefinition;
+import tc.oc.pgm.api.feature.FeatureInfo;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.util.TimeUtils;
+import tc.oc.pgm.util.compose.Composition;
+
+@FeatureInfo(name = "fill")
+public class FillerDefinition implements FeatureDefinition {
+
+  private final Composition<ItemStack> loot;
+  private final Filter filter;
+  private final @Nullable Filter refillTrigger; // Dynamic
+
+  private final boolean usesRefillTrigger;
+  private final Duration refillInterval;
+
+  private final boolean usesRefillInterval;
+  private final boolean clearBeforeRefill;
+
+  public FillerDefinition(
+      Composition<ItemStack> loot,
+      Filter filter,
+      @Nullable Filter refillTrigger,
+      Duration refillInterval,
+      boolean clearBeforeRefill) {
+    this.loot = loot;
+    this.filter = filter;
+    this.refillTrigger = refillTrigger;
+    this.refillInterval = refillInterval;
+    this.clearBeforeRefill = clearBeforeRefill;
+    this.usesRefillTrigger = this.refillTrigger != null;
+    this.usesRefillInterval = this.refillInterval != TimeUtils.INFINITE_DURATION;
+  }
+
+  /** Items to fill with */
+  Composition<ItemStack> getLoot() {
```

Comment:
```suggestion
  public Composition<ItemStack> getLoot() {
```
make all of the methods here public. I assume you copied them from legacy pgm where they were interfaces, which made them public by default

Reply:
you got me there